cmake/Corrosion.cmake

```diff
@@ -1142,6 +1142,12 @@ function(corrosion_set_hostbuild target_name)
     )
 endfunction()
 
+function(corrosion_add_target_generated_headers target_name)
+    set(generated_header_dir "${CMAKE_CURRENT_BINARY_DIR}/generated-code")
```

Comment:
ah right, CMAKE_CURRENT_BINARY_DIR only creates separate directories for *CMake* targets; it doesn't help to separate outputs from multiple crates linked into one CMake target

Reply:
> ah right, CMAKE_CURRENT_BINARY_DIR only creates separate directories for CMake targets

`CMAKE_CURRENT_BINARY_DIR` corresponds to the current list dir (just relative in the build tree instead of the source tree). CMake targets are kind of orthogonal to that, but most of the time you do define different targets in different list directories.